src/toil/job.py

```diff
@@ -1154,17 +1154,33 @@ def readGlobalFile(self, fileStoreID, userPath=None, cache=True, mutable=None):
                         self.returnFileSize(fileStoreID, localFilePath, lockFileHandle,
                                             fileAlreadyCached=True)
                 # If the file is not in cache, check whether the .harbinger file for the given
-                # FileStoreID exists. If it does, the wait and periodically check for the removal of
-                # the file and the addition of the completed download into cache of the file by the
-                # other job. Then we link to it.
+                # FileStoreID exists.  If it does, the wait and periodically check for the removal
+                # of the file and the addition of the completed download into cache of the file by
+                # the other job. Then we link to it.
                 elif fileIsLocal and os.path.exists(harbingerFileName):
                     logger.info('CACHE: Waiting for another worker to download file with ID %s.'
                                 % fileStoreID)
                     while os.path.exists(harbingerFileName):
-                        # Release the file lock and then periodically check for completed download
-                        flock(lockFileHandle, LOCK_UN)
-                        time.sleep(20)  # What should this value be?
-                        flock(lockFileHandle, LOCK_EX)
+                        # Ensure that the process downloading the file is still alive.  The PID will
+                        # be in the harbinger file.
+                        try:
+                            pid = int(open(harbingerFileName).read())
+                        except ValueError:
+                            # There's a chance the downloading process died before it could write
+                            # into the harbinger file
```

Comment:
Why is this still here? With the atomic creation you added, it is highly unlikely that you'll get a ValueError, correct? If so, I would rather let the ValueError through.